src/pages/FightPage/FightPage.tsx

```diff
@@ -7,43 +7,72 @@ import {
   FightButtonStyle,
   FightActionWrapper,
   ActiveButtonStyle,
-  DisabledButtonStyle,
 } from "./styles";
 import { HeadingMediumRegular } from "../../styles/typography";
 import Autocomplete from "../../components/Autocomplete/Autocomplete";
 import PokemonFightCard from "../../features/fight/PokemonFightCard/PokemonFightCard";
 import pokemonsMockData from "../../data/pokemonMockData";
 import strengthIcon from "../../assets/icons/strength.svg";
 import Button from "../../components/Button/Button";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import { IPokemonData } from "../../types/pokemonTypes";
-import { transformPokemonDataToOption } from "../../utils/transformDataToOption";
+import { transformPokemonDataToOption } from "../../utils/transformData";
+import {
+  calculateCatchRate,
+  calculateDamage,
+  getRandomPokemon,
+} from "../../utils/fightPageFunctions";
 
-const getRandomPokemon = () => {
-  return pokemonsMockData[Math.floor(Math.random() * pokemonsMockData.length)];
-};
+type PlayerTurn = "player" | "opponent";
+
+const myPokemonsData = pokemonsMockData.filter((pokemon) => pokemon.isOwned);
 
 const FightPage = () => {
   const [isActiveFight, setIsActiveFight] = useState(false);
-  const [canCatch, setCanCatch] = useState(false); // will be used to enable/disable catch button based on backend
+  const [canCatch, setCanCatch] = useState(false);
   const [selectedPokemon, setSelectedPokemon] = useState<IPokemonData>(
-    getRandomPokemon()
+    getRandomPokemon(myPokemonsData)
   );
-  const [opponentPokemon, setOpponentPokemon] = useState<IPokemonData>(
-    getRandomPokemon()
+  const [opponentPokemon] = useState<IPokemonData>(
+    getRandomPokemon(pokemonsMockData, selectedPokemon)
+  );
+  const [selectedPokemonHp, setSelectedPokemonHp] = useState(
```

Comment:
need both `selectedPokemonHp` and `selectedPokemon`?
is `selectedPokemonHp` current hp? if yes, change the name of the state